airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/json_schema_helper.py

```diff
@@ -86,3 +87,59 @@ def get_property(self, path: List[str]) -> Mapping[str, Any]:
 
     def field(self, path: List[str]) -> CatalogField:
         return CatalogField(schema=self.get_property(path), path=path)
+
+    def find_variant_pathes(self) -> List[List[str]]:
+        """
+        return list of json object pathes for oneOf or anyOf attributes
+        """
+        variant_pathes = []
+
+        def traverse_schema(_schema, path=[]):
+            if path and path[-1] in ["oneOf", "anyOf"]:
+                variant_pathes.append(path)
+            for item in _schema:
+                next_obj = _schema[item] if isinstance(_schema, dict) else item
+                if isinstance(next_obj, (list, dict)):
+                    traverse_schema(next_obj, [*path, item])
+
+        traverse_schema(self._schema)
+        return variant_pathes
+
+    def validate_variant_pathes(self, variant_pathes: List[List[str]]):
+        """
+        Validate oneOf pathes according to reference
+        https://docs.airbyte.io/connector-development/connector-specification-reference
+        """
+
+        def get_top_level_item(variant_path: List[str]):
+            # valid path should contain at least 3 items
+            path_to_schema_obj = variant_path[:-1]
+            return dpath.util.get(self._schema, "/".join(path_to_schema_obj))
+
+        for variant_path in variant_pathes:
+            top_level_obj = get_top_level_item(variant_path)
+            if "$ref" in top_level_obj:
+                obj_def = top_level_obj["$ref"].split("/")[-1]
+                top_level_obj = self._schema["definitions"][obj_def]
+            """
+            1. The top-level item containing the oneOf must have type: object
+            """
+            assert top_level_obj.get("type") == "object"
```

Comment:
If you think of the developer experience, it really helps them understand what's wrong if the assertions display a helpful error message e.g: 

```suggestion
            assert top_level_obj.get("type") == "object", f"The top-level definition in a `oneOf` block should have type: object. misconfigured object: {top_level_obj}. See specification reference at https://docs.airbyte.io/connector-development/connector-specification-reference`
```

Can you add a link to https://docs.airbyte.io/connector-development/connector-specification-reference in all the assert messages so the user knows where to find the reference?

Reply:
Can you also add a message to the assert below?